### ISSUE TITLE
Simple feature request: transformers for continuous inputs

### ISSUE THREAD

**Issue by guillefix** [lucidrains/x-transformers#32]
I think it would be useful to add an option to the TransformerWrapper, or perhaps to make a new Wrapper type, that does not use embedding layers, so that the inputs are real-valued vectors. This would allow to use x-transformers for tasks with continuous inputs. For example [here](https://arxiv.org/abs/2101.08779) they use transformers in that way, and also most of the times transformers are applied to regression tasks.

Note that in this case, one then just talks about input and output dimension, and not number of tokens.

Perhaps the cleanest way to do this is to make a new Wrapper type that works with continuous vector inputs, and then make TransformerWrapper use this ContinuousTransformerWrapper inside it, with input dimension being the embeding dimension, and output dimension being num_tokens. Hope this makes sense!

**Comment by lucidrains** [comment]
@guillefix yup, makes sense, but couldn't you just use `Encoder` or `Decoder` directly?

**Comment by lucidrains** [comment]
@guillefix a short code snippet of what you are thinking of would help!

**Comment by guillefix** [comment]
Ah, you are right. I guess I didn't look at what Encoder/Decoder did. Tho I guess the wrapper may still be useful, to have the AbsolutePositionalEmbedding, dropout, and linear layers, conveniently packaged inside.

I want to have something like the BERT full-attention encoder (which is what AIST++ use), but with continuous inputs

```
import torch
from x_transformers import ContinuousTransformerWrapper, Encoder

model = ContinuousTransformerWrapper(
    dim_in = 63,
	dim_out = 100,
    max_seq_len = 1024,
    attn_layers = Encoder(
        dim = 512,
        depth = 12,
        heads = 8
    )
).cuda()

x = torch.randn((1, 1024, 63)).cuda() # or perhaps #(1024,1,63) to have same input shape as pytorch's transformers?
mask = torch.ones_like(x).bool()

model(x, mask = mask) # (1, 1024, 100)
```

Tho I'd understand if this is too minimal a change/niche to be included, but it would mean a drop-in replacement in my implementation of AIST++

**Comment by lucidrains** [comment]
yup, i could do that, although if you use the new rotary embeddings (`rotary_pos_emb = True` on the encoder), you will get vastly better results

**Comment by lucidrains** [comment]
i'll see what i can do tomorrow morning!

**Comment by guillefix** [comment]
Regarding rotary embedings, yeah, that's one of the reasons why I wanted to try x-transformers! I was just showing a vanilla example above

**Comment by lucidrains** [comment]
@guillefix yup! with rotary embeds, you wouldn't even need the absolute positional embedding
